UCSLibraries/src/main/java/it/cnr/iit/xacml/wrappers/PolicyWrapper.java

```diff
@@ -68,44 +68,55 @@ public static PolicyWrapper build( String policy ) {
 
         PolicyWrapper policyWrapper = new PolicyWrapper();
         policyWrapper.setPolicy( policy );
+        policyWrapper.setPolicyType( unmarshalPolicyType( policy ) );
 
-        return policyWrapper.policyType != null ? policyWrapper : null;
+        Reject.ifNull( policyWrapper.getPolicyType() );
+
+        return policyWrapper;
     }
 
-    @Override
-    public List<Attribute> getAttributesForCondition( String conditionName ) {
-        Reject.ifBlank( conditionName );
+    public static PolicyWrapper build( PolicyType policyType ) {
+        Reject.ifNull( policyType );
 
-        List<Object> list = policyType.getCombinerParametersOrRuleCombinerParametersOrVariableDefinition();
+        PolicyWrapper policyWrapper = new PolicyWrapper();
+        policyWrapper.setPolicyType( policyType );
+        policyWrapper.setPolicy( marshalPolicyType( policyType ) );
 
-        for( Object obj : list ) {
-            if( !( obj instanceof RuleType ) ) {
-                continue;
-            }
+        Reject.ifBlank( policyWrapper.getPolicy() );
```

Comment:
Do you really need this? Because in line 79 you test policyType, can marshalling return null?

Reply:
Yes since marshalling can return null for now it's needed.
It's very ugly I'll change it by handling the exceptions in those builders / sort of copy constructors :P (also ugly)